### DISCUSSION TITLE
Features for sda-dropwizard-commons 6

### DISCUSSION THREAD

**Discussion by christopher-cudennec** [SDA-SE/sda-dropwizard-commons#2435] [Ideas]
Happy to get your opinions on the features that should be provided by the next major version. I will update this list from time to time.

# High probablility

* Java versions
  * drop support for Java 11
  * add support for Java 21 (LTS) (due in September 2023)
* update [Dropwizard](https://github.com/dropwizard/dropwizard) to version 4
  * jakarta hassle
* update resilience4j to version 2.0
  * needs Java 17
* update to Spring Data Mongo 4.x
  * gets rid of warning regarding Joda Time converters
  * jakarta hassle
  * upgrades for MongoDB server 6
* update Flapdoodle / rewrite sda-commons-server-mongo-testing
* update json-unit to version 3.0
  * needs Java 17
* re-use async-api documentation from sda-spring-boot-commons
  * use dependency to [sda-commons-asyncapi](https://github.com/SDA-SE/sda-spring-boot-commons/blob/spring-boot-3/sda-commons-asyncapi/build.gradle)?
* update to [Weld 5](https://weld.cdi-spec.org/news/) 

# Up for discussion

* rework the EnabledFieldFilter annotation
  * completely remove it?
  * keep it
  * … and add it to the OpenAPI documentation automatically?
  * … and provide a way to retrieve the fields to also consider them in the database queries
  * 2023-06-14: The field filter was very valuable for one of our products. At the moment I would vote to keep it.
* Kafka Bundle
  * can we refactor the health check to get rid of the admin client?
  * can we simplify creating consumers / producers?
  * do we want to use the “official” [dropwizard-kafka](https://github.com/dropwizard/dropwizard-kafka)?
* Remove `http_proxy` to close https://github.com/SDA-SE/sda-dropwizard-commons/issues/2336

**Comment by JoergSiebahn** [comment]
Hold back [major update of json-unit](https://github.com/SDA-SE/sda-dropwizard-commons/pull/2506) because Java 11 support was dropped and a minor breaking change in the API.

We should keep in mind to do the upgrade with the next major release.

**Reply by christopher-cudennec** [reply]
updated the feature list above 👍

**Comment by sliesche** [comment]
KafkaBundle [checks initialisation](https://github.com/SDA-SE/sda-dropwizard-commons/blob/504515abe1fef4068e9d39ed5289454f5ee1b697/sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/KafkaBundle.java#L434) only in some public methods
Should we check in all public createConsumer() , createProducer() methods for initialisation?

**Comment by christopher-cudennec** [comment]
Closing discussion as version 6 has just been released. 🥳
